### ISSUE TITLE
Assert failed for ImGui::InputTextMultiline

### ISSUE THREAD

**Issue by GreedyTactician** [ocornut/imgui#5195]
![Screenshot from 2022-04-13 13-24-33](https://user-images.githubusercontent.com/18636088/163236260-a4eee1e2-40c3-4c49-9d8e-20bde8a6d3db.png)

buf is NULL
buf_size is 1

Note: I am using the ImGui::InputTextMultiline from imgui_stdlib.h

This is the line of code:
`ImGui::InputTextMultiline("##source", shownText, ImVec2(-FLT_MIN, -1), flags, NULL, NULL);`

shownText looks like a normal empty string before I pass it in.

I am also using imnodes (https://github.com/Nelarius/imnodes). Why do I mention this? Because the bug happens when I interact with the input pin of the first node I create and then create a new node. If I don't play with the pin, the assert doesn't fail. If I create more than one node and then play with the input pin, the assert doesn't fail.

What does this have to ImGui::InputTextMultiline and shownText? I have no idea. This is strange.

**Comment by ocornut** [comment]
> Its insufferable to deal with your logic sometimes :(

Maybe there's an ID conflict somewhere?

**Comment by GreedyTactician** [comment]
"Its insufferable to deal with your logic sometimes :(" where does this quote come from and what does it refer to?

As for the ID conflict, I am not too sure how to investigate this. I don't know what calls TextInputEx. I don't know the purpose of TextInputEx. I have no idea how TextInputEx could be related to imnodes. I don't know if the faulty code lies in imgui or imnodes, but I am pretty sure it is not my code.

**Comment by ocornut** [comment]
(Oops I am so sorry we were arguing with a good friend and pasted a totally unrelated quote. So sorry)

Try to change `##source` to anything else prefixed with ## just to see if it makes a difference so you can at least clear that hypothesis.

**Comment by GreedyTactician** [comment]
Perhaps I should bring this issue to imnodes' github instead? I posted it here since the assertion that failed is in imgui so perhaps you have an idea what step went wrong before that assertion.

**Comment by GreedyTactician** [comment]
> (Oops I am so sorry we were arguing with a good friend and pasted a totally unrelated quote. So sorry)
> 
> Try to change `##source` to anything else prefixed with ## just to see if it makes a difference so you can at least clear that hypothesis.

Sure thing. I'll be home in 3 hours so I'll try that then.

**Comment by ocornut** [comment]
> shownText looks like a normal empty string before I pass it in.

Can you check this is the case in the callstack when the assert triggered?

> buf is NULL
> buf_size is 1

buf is passed from .c_str() in `imgui_stdlib.cpp`. As far as I understand, std::string::c_str() by specs should never return NULL.
I would imagine an accidentally overwrite your string instance and/or having it destructed could lead to that. Right now with what I am seeing I can't understand how there can be an issue on your side, but I might be overlooking a clue.

**Comment by GreedyTactician** [comment]
I fixed it. The error was in my code. 

Pins were a red herring (when I selected the pins, I actually selected the node too most of the time). The bug was my lack of awareness of the fundamentals of vectors. When vectors grow, stuff gets moved around and pointers to that stuff don't work anymore (I kind of knew that but it didn't 'click' as I was typing code). 

I am new to C++ :)

Thank you for your time and assistance
